src/main/java/org/minions/devfund/reinaldo/WhackAMole.java

```diff
@@ -0,0 +1,184 @@
+package org.minions.devfund.reinaldo;
+
+/**
+ * Created by reinaldo on 02-07-18.
+ */
+public class WhackAMole {
+
+    private static final char DEFAULTCHARACTER = '*';
+    private static final String NEWLINE = "\n";
+    private static final String SPACES = "  ";
+    private static final char WHACKED = 'W';
+    public static final char MOLE = 'M';
+
+    //instance variables called score, molesLeft, and attemptsLeft.
+    // Make one more instance variable called moleGrid which is a 2-dimensional array of chars.
+    private int score;
+    private int molesLeft;
+    private int attemptsLeft;
+    private char[][] moleGrid;
+    private int gridDimension;
+
+    /**
+     * The methods to get the mole grid.
+     * @return the char [][]
+     */
+    public char[][] getMoleGrid() {
+        return moleGrid.clone();
+    }
+
+    /**
+     * The methods to get the moles left.
+     * @return the number of moles.
+     */
+    public int getMolesLeft() {
+        return molesLeft;
+    }
+
+    /**
+     * Gets the score value.
+     * @return int
+     */
+    public int getScore() {
+        return  score;
+    }
+
+    /**
+     * Gets the AttemptsLeft.
+     * @return int
+     */
+    public int getAttemptsLeft() {
+        return attemptsLeft;
+    }
+
+    /**
+     *
+     *
+     Constructor for the game, specifies total number of whacks allowed, and the grid dimension.
+     *
+     *
+     * @param numAttempts numero of attempts.
+     * @param gridDimension the arrays.
+     */
+    public WhackAMole(int numAttempts, int gridDimension) {
+        attemptsLeft = numAttempts;
+        this.gridDimension = gridDimension;
+        createTheMoleGrid();
+    }
+
+    /**
+     * Methods to fill the mole grid for the game whack a mole.
+     *
+     */
+    public void createTheMoleGrid() {
+        moleGrid = new char[gridDimension][gridDimension];
+        for (int i = 0; i < gridDimension; i++) {
+            for (int j = 0; j < moleGrid[i].length; j++) {
+                moleGrid[i][j] = DEFAULTCHARACTER;
+            }
+        }
+    }
+
+    /**
+     * This is the location for a wack a mole.
+     * @param posX position in x.
+     * @param posY position.
+     * @return true or false in the location from (x,y).
+     */
+    public boolean place(int posX, int posY) {
+        if (moleGrid[posX][posY] != MOLE) {
+            moleGrid[posX][posY] = MOLE;
+            molesLeft++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * location, take a whack at that location. If that location.
+     * @param posX is the location.
+     * @param posY is the location
+     */
+    public void whack(int posX, int posY) {
+        if (moleGrid[posX][posY] == MOLE) {
+            updateForScoreMolesLeft();
+            updateForTheGrid(posX, posY);
+            return;
+        }
+        updateForTheGrid(posX, posY);
```

Comment:
In this point is really needed to do this
moleGrid[posX][posY] = WHACKED;
?
or only you should do this
attemptsLeft--;
?

Reply:
It is necesary in the two case the moleGrid[posX][posY] = WHACKED  for the use the print methods